src/mono/mono/mini/mini-exceptions.c

```diff
@@ -3500,9 +3500,14 @@ mini_llvmonly_throw_corlib_exception (guint32 ex_token_index)
 	guint32 ex_token = MONO_TOKEN_TYPE_DEF | ex_token_index;
 	MonoException *ex;
 
+	HANDLE_FUNCTION_ENTER ();
+
 	ex = mono_exception_from_token (m_class_get_image (mono_defaults.exception_class), ex_token);
+	MONO_HANDLE_PIN (ex);
 
 	mini_llvmonly_throw_exception ((MonoObject*)ex);
+
+	HANDLE_FUNCTION_RETURN ();
```

Comment:
`mini_llvmonly_throw_exception` starts stack unwinding, so the cleanup in `HANDLE_FUNCTION_RETURN` won't happen and we'll leak a coop handle.

Can you pass `&ex` to `mini_llvmonly_throw_exception` wouldn't that be enough to pin `ex` on the stack until the stack unwinding starts?

Reply:
Stored it in jit_tls instead.